### PR TITLE
set value to characteristic after reading it

### PR DESCRIPTION
**Pull request by lpapazow**
I made the changes in BlueGigaBluetoothDevice.java because after calling `device.readCharacteristic()` method in my handler and calling `myCharacteristic.getValue()`, I get an empty array and I also saw that none of the overloaded `BluetoothCharacteristic.setValue()` method implementations was used in the BlueGiga Bundle to set the value after reading it from the device. These changes seem to have worked for my device (a distance sensor).

I am suggesting the changes in the BeaconBluetoothHandler.java, because I noticed that when trying to initialize the device with BlueZ, the `thing.configuration()`  doesn't have **address** as a key, but `thing.getProperties()` does. My DiscoveryParticipant implementation creates the DiscoveryResult with the address included in the properties.

Signed-off-by: Lyubomir V. Papazov <lpapazow@gmail.com>

### CODE REVIEW

File: extensions/binding/org.eclipse.smarthome.binding.bluetooth.bluegiga/src/main/java/org/eclipse/smarthome/binding/bluetooth/bluegiga/BlueGigaBluetoothDevice.java [hunk 1]
```diff
@@ -417,6 +417,11 @@ public void bluegigaEventReceived(BlueGigaResponse event) {
             if (characteristic == null) {
                 logger.debug("BlueGiga didn't find characteristic for event {}", event);
             } else {
+
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.bluetooth.bluegiga/src/main/java/org/eclipse/smarthome/binding/bluetooth/bluegiga/BlueGigaBluetoothDevice.java]: could you remove this empty line?

File: extensions/binding/org.eclipse.smarthome.binding.bluetooth.bluegiga/src/main/java/org/eclipse/smarthome/binding/bluetooth/bluegiga/BlueGigaBluetoothDevice.java [hunk 2]
```diff
@@ -417,6 +417,11 @@ public void bluegigaEventReceived(BlueGigaResponse event) {
             if (characteristic == null) {
                 logger.debug("BlueGiga didn't find characteristic for event {}", event);
             } else {
+            	
+            	 procedureCharacteristic.setValue(valueEvent.getValue());
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.bluetooth.bluegiga/src/main/java/org/eclipse/smarthome/binding/bluetooth/bluegiga/BlueGigaBluetoothDevice.java]: could you remove the superfluous space?

File: extensions/binding/org.eclipse.smarthome.binding.bluetooth.bluegiga/src/main/java/org/eclipse/smarthome/binding/bluetooth/bluegiga/BlueGigaBluetoothDevice.java [hunk 3]
```diff
@@ -417,6 +417,11 @@ public void bluegigaEventReceived(BlueGigaResponse event) {
...
+            	// Notify the user of the updated value
+            	notifyListeners(BluetoothEventType.CHARACTERISTIC_UPDATED, procedureCharacteristic);
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.bluetooth.bluegiga/src/main/java/org/eclipse/smarthome/binding/bluetooth/bluegiga/BlueGigaBluetoothDevice.java]: Why did you move this code up here? This changes the order of the notifications and imho it was just fine before.

File: extensions/binding/org.eclipse.smarthome.binding.bluetooth/src/main/java/org/eclipse/smarthome/binding/bluetooth/BeaconBluetoothHandler.java [hunk 4]
```diff
@@ -47,7 +48,7 @@ public BeaconBluetoothHandler(@NonNull Thing thing) {
     @Override
     public void initialize() {
         try {
-            address = new BluetoothAddress(getConfig().get(BluetoothBindingConstants.CONFIGURATION_ADDRESS).toString());
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.bluetooth/src/main/java/org/eclipse/smarthome/binding/bluetooth/BeaconBluetoothHandler.java]: Please revert this change - the address is defines as a configuration parameter, so it must exist for this handler.
We rather need to figure out why you have the situation that you have Things that only have the address set as a property, but not as a configuration parameter.

**Review by kaikreuzer — CHANGES_REQUESTED** [review]

File: extensions/binding/org.eclipse.smarthome.binding.bluetooth/src/main/java/org/eclipse/smarthome/binding/bluetooth/BeaconBluetoothHandler.java (hunk 4 above)
Comment (lpapazow) [reply extensions/binding/org.eclipse.smarthome.binding.bluetooth/src/main/java/org/eclipse/smarthome/binding/bluetooth/BeaconBluetoothHandler.java]: OK, I will try to see why my Thing doesn't have the address in it's configuration.

File: extensions/binding/org.eclipse.smarthome.binding.bluetooth.bluegiga/src/main/java/org/eclipse/smarthome/binding/bluetooth/bluegiga/BlueGigaBluetoothDevice.java (hunk 3 above)
Comment (lpapazow) [reply extensions/binding/org.eclipse.smarthome.binding.bluetooth.bluegiga/src/main/java/org/eclipse/smarthome/binding/bluetooth/bluegiga/BlueGigaBluetoothDevice.java]: I thought the *CHARACTERISTIC_UPDATED* event should send the `procedureCharacteristic` before it is set to null [here](https://github.com/kaikreuzer/smarthome/pull/8/files#diff-a7b7ef841bb09679f68c5cab00300b0bR426) as I used the characteristic in the `onCharacteristicUpdate` method in my binding initially. I don't override the `onCharacteristicUpdate` in my binding anymore and I reverted the change, but I'm still curious why would the code [here](https://github.com/kaikreuzer/smarthome/pull/8/files#diff-a7b7ef841bb09679f68c5cab00300b0bL430) pass null as the second argument in case of a successful reading.
